src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int START_POSITION = 0;
+    private static final double CAPACITY_INCREASE = 1.5;
+    private int size;
+    private T[] elements;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void ensureCapacity() {
+        if (size == elements.length) {
+            int newCapacity = (int) (size * CAPACITY_INCREASE);
+            T[] newElements = (T[]) new Object[newCapacity];
+            System.arraycopy(elements, START_POSITION, newElements, START_POSITION, size);
+            elements = newElements;
+        }
```

Comment:
The logic for resizing the array should be extracted into a separate method to adhere to the Single Responsibility Principle.